packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -483,7 +483,7 @@ class CupertinoTextField extends StatefulWidget {
     properties.add(DiagnosticsProperty<bool>('expands', expands, defaultValue: false));
     properties.add(IntProperty('maxLength', maxLength, defaultValue: null));
     properties.add(FlagProperty('maxLengthEnforced', value: maxLengthEnforced, ifTrue: 'max length enforced'));
-    properties.add(DiagnosticsProperty<Color>('cursorColor', cursorColor, defaultValue: null));
```

Comment:
@pq with this change we would need to add a lint to warn if DiagnosticsProperty<Color> is used instead of ColorDiagnosticsProperty.